flume-ng-core/src/main/java/org/apache/flume/source/AvroSource.java

```diff
@@ -300,28 +306,30 @@ public ChannelPipeline getPipeline() throws Exception {
   public void stop() {
     logger.info("Avro source {} stopping: {}", getName(), this);
 
-    server.close();
+    if (server != null) {
+      server.close();
+      try {
+        server.join();
+        server = null;
+      } catch (InterruptedException e) {
+        logger.info("Avro source " + getName() + ": Interrupted while waiting " +
```

Comment:
I think the`Thread.currentThread().interrupt();` should be called here to set the interrupt flag of the thread.